### PR TITLE
Adding docs for new Influx 2.0 capabilities

### PR DESCRIPTION
**Pull request by mdegat01**
Documentation additions to go along with [Influx 2.0 PR](https://github.com/home-assistant/core/pull/35392).  Also small adjustments to the 1.xx documentation to mention the fact that the `where` option in the sensor supports templates and add a new `database` option to the sensor platform to allow you to change the default db for all sensors.

## Proposed change

This is a corresponding documentation update to along with the [Influx 2.0 PR](https://github.com/home-assistant/core/pull/35392).  See details on that PR for reasoning of the new feature addition.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/35392
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/influxdb.markdown
Comment (klaasnicolaas) [inline source/_integrations/influxdb.markdown]: If you start the description with a `*`, you need to quote from begin to end with `" "`

```suggestion
  description: "**1.xx only** IP address of your database host, e.g., 192.168.1.10"
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/influxdb.markdown
Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: Fixed. And thank you!  That error message was driving me crazy, I did not know what it was talking about with the line 4 column 16 bit.

File: source/_integrations/influxdb.markdown
Comment (frenck) [inline source/_integrations/influxdb.markdown]: So what if v3 comes out? This feels confusing. I feel that there should be a single option which takes a version as a parameter, defaulting to v1.

File: source/_integrations/influxdb.markdown
Comment (frenck) [inline source/_integrations/influxdb.markdown]: Can't this be merged with the host/port configuration of v1?
Basically, even v1 API is an URL endpoint

Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: Yep you're right, modified the two PRs accordingly.  `url` option is gone, building the URL from the other config options.

File: source/_integrations/influxdb.markdown
Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: The thread [here](https://github.com/home-assistant/core/pull/35392#pullrequestreview-421997905) has the reasoning behind this. I agree with the logic and at first I tried to implement it this way but discovered if I switch to a version field that takes a number there's a validation issue I'm not sure how to deal with.  Waiting on a second opinion in the other PR there to see how to best resolve this.

Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: @frenck Issue should be resolved now. A solution to the validation issue was found so now both doc and runtime are updated to use an `api_version` config option instead of the v2 specific `api_v2`.

File: source/_integrations/influxdb.markdown
Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: @klaasnicolaas this issue should be resolved now but I can't seem to mark it as such, still showing up as a requested change blocking merge. Can you resolve this one? Just trying to get everything up for a merge now that the implementation is nearing completion.

**Comment by mdegat01** [comment]
Implementation PR has now been merged. Please let me know if any other changes are needed here so the doc can go in as well.

File: source/_integrations/influxdb.markdown
Comment (frenck) [inline source/_integrations/influxdb.markdown]: This does not need emphasis, just write it in the normal text would do.

The same applies to all other occasions as well.

File: source/_integrations/influxdb.markdown
Comment (frenck) [inline source/_integrations/influxdb.markdown]: Why is this repeated with all options again?

I'm confused... a newer user would be as well I guess.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/influxdb.markdown
Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: Ok will remove the emphasis.

File: source/_integrations/influxdb.markdown
Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: My thinking was that it was different enough for 2.xx with the entirely different query language that it would be easier to just separate 1.xx and 2.xx configurations and examples. But I can merge and just put in `2.xx only` for things like I did other places.

Comment (mdegat01) [reply source/_integrations/influxdb.markdown]: Ok merged the config sections, let me know what you think

**Review by frenck — APPROVED** [review]
Thanks, @mdegat01 👍

**Pull request merged**
